lib/bolt/applicator.rb

```diff
@@ -134,12 +135,26 @@ def apply(args, apply_body, scope)
 
       targets = @inventory.get_targets(args[0])
 
-      ast = Puppet::Pops::Serialization::ToDataConverter.convert(apply_body, rich_data: true, symbol_to_string: true)
+      apply_ast(apply_body, targets, options, plan_vars)
+    end
 
-      apply_ast(ast, targets, options, plan_vars)
+    # Count the number of top-level statements in the AST.
+    def count_statements(ast)
+      case ast
+      when Puppet::Pops::Model::Program
+        count_statements(ast.body)
+      when Puppet::Pops::Model::BlockExpression
+        ast.statements.count
+      else
+        1
+      end
     end
 
-    def apply_ast(ast, targets, options, plan_vars = {})
+    def apply_ast(raw_ast, targets, options, plan_vars = {})
```

Comment:
It seems like this will report analytics for bolt apply as well as apply statements. I'm not sure if thats intentional. We may want to track those separately but I'm not sure.

Reply:
That was intentional. I could change it if we do want to track them separately.